pkg/controllers/deprovisioning/controller.go

```diff
@@ -115,32 +116,13 @@ func (c *Controller) Reconcile(ctx context.Context, _ reconcile.Request) (reconc
 	// Attempt different deprovisioning methods. We'll only let one method perform an action
 	isConsolidated := c.cluster.Consolidated()
 	for _, d := range c.deprovisioners {
-		candidates, err := GetCandidates(ctx, c.cluster, c.kubeClient, c.clock, c.cloudProvider, d.ShouldDeprovision)
+		success, err := c.deprovision(ctx, d)
 		if err != nil {
-			return reconcile.Result{}, fmt.Errorf("determining candidates, %w", err)
+			logging.FromContext(ctx).Infof("could not deprovision with %q, %s", d, err)
```

Comment:
This feels strange to me. Why are we logging the error as info and not handling error returns with a standard rate limiter like we would do through controller-runtime?

Reply:
We don't want exponential backoff if some deprovisioner fails.  E.g. if expiration starts failing for some reason, we don't want to pause all consolidation

Reply:
Right, you can configure the RateLimiter to not do exponential backoff but instead have no backoff or have a consistent rate that it requeues at. It seems odd that we wouldn't just be utilizing built-in error logging from controller-runtime.

Reply:
We shouldn't be using the controller-runtime for reconciling non-existent objects.  That single change to deprovisioning has introduced two separate bugs. I don't want to add more controller-runtime specific stuff to this so that undoing that change is easier.